doc/tutorials/using_vxworks_as_uos.rst

```diff
@@ -0,0 +1,139 @@
+   to setup the VxWorks development environment and build the VxWorks Image.
+
+   .. note::
+      The following kernel configuration should be excluded:
+        * INCLUDE_PC_CONSOLE
+        * DRV_SIO_PCI_NS16550
+        * SELECT_PC_CONSOLE_KBD
+
+      The following kernel configuration should be included:
+        * DRV_SIO_IA_NS16550
+        * DRV_SIO_FDT_NS16550
+
+      The following kernel configuration value should be set:
+        * CONSOLE_BAUD_RATE = 115200
+        * SYS_CLK_RATE_MAX = 1000
+
+#. Build GRUB2 BootLoader Image
+
+   We use grub-2.02 as the bootloader of VxWorks in this tutorial. You may gave other version of GRUB2 a try.
+
+   .. code-block:: none
+
+      $ curl ftp://ftp.gnu.org/gnu/grub/grub-2.02.tar.xz | tar xJv
+      $ cd grub-2.02
+      $ ./autogen.sh
+      $ ./configure --with-platform=efi --target=x86_64
+      $ make
+      $ ./grub-mkimage -p /EFI/BOOT -d ./grub-core/ -O x86_64-efi -o grub_x86_64.efi \
+            boot efifwsetup efi_gop efinet efi_uga lsefimmap lsefi lsefisystab \
+            exfat fat multiboot2 multiboot terminal part_msdos part_gpt normal \
+            all_video aout configfile echo file fixvideo fshelp gfxterm gfxmenu \
+            gfxterm_background gfxterm_menu legacycfg video_bochs video_cirrus \
+            video_colors video_fb videoinfo video net tftp
+
+   Then we will get ``grub_x86_64.efi`` in current directory.
+
+#. Preparing the Boot Device
+
+   .. code-block:: none
+
+      $ dd if=/dev/zero of=VxWorks.img bs=1M count=35
+      $ mkfs.vfat -F 32 VxWorks.img
+      $ sudo mount `sudo losetup -f -P --show VxWorks.img` /mnt
+
+   Create the following directories.
+
+   .. code-block:: none
+
+      $ sudo mkdir -p /mnt/efi/boot
+      $ sudo mkdir -p /mnt/kernel
+
+   Copy ``vxWorks`` and ``grub_x86_64.efi``.
+
+   .. code-block:: none
+
+      $ sudo cp vxWorks /mnt/kernel/
+      $ sudo cp grub-2.02/grub_x86_64.efi /mnt/efi/boot/bootx64.efi
+
+   Create ``/mnt/efi/boot/grub.cfg`` containing the following:
+
+   .. code-block:: console
+
+      set default=0
+      set timeout=5
+
+      menuentry "VxWorks Guest" {
+          multiboot /kernel/vxWorks
+      }
+
+   Unmount the loopback device:
+
+   .. code-block:: none
+
+      $ sudo umount /mnt
+
+   You now have a virtual disk image with bootable VxWorks in ``VxWorks.img``. If the VxWorks build system
+   is not the ACRN SOS, then you will need to transfer this image to ACRN SOS (via, e.g, a USB stick or network)
+
+#. Follow :ref:`getting-started-apl-nuc` to boot "The ACRN Service OS".
+
+#. Boot VxWorks as User OS.
+
+   On the ACRN SOS, prepare a directory and populate it with VxWorks files.
+
+   .. code-block:: none
+
+      $ mkdir vxworks && cd vxworks
+      $ cp /usr/share/acrn/samples/nuc/launch_vxworks.sh .
+      $ cp /usr/share/acrn/bios/OVMF.fd .
+
+   You will also need to copy the ``VxWorks.img`` created in the above section into directory ``vxworks``.
```

Comment:
You didn't provide the command to do this (as you did for all the other steps).

Reply:
This image locates in other machine. We can transfer it to the ACRN SOS through a USB stick or network. But it is hard to provide the command to do this.